pkg/rpc/server/server.go

```diff
@@ -548,6 +550,86 @@ func (s *Server) validateAddress(reqParams request.Params) (interface{}, *respon
 	return validateAddress(param.Value), nil
 }
 
+// calculateNetworkFee calculates network fee for the transaction.
+func (s *Server) calculateNetworkFee(reqParams request.Params) (interface{}, *response.Error) {
+	if len(reqParams) < 1 {
+		return 0, response.ErrInvalidParams
+	}
+	byteTx, err := reqParams[0].GetBytesBase64()
+	if err != nil {
+		return 0, response.WrapErrorWithData(response.ErrInvalidParams, err)
+	}
+	tx, err := transaction.NewTransactionFromBytes(s.network, byteTx)
+	if err != nil {
+		return 0, response.WrapErrorWithData(response.ErrInvalidParams, err)
+	}
+	hashablePart, err := tx.EncodeHashableFields()
+	if err != nil {
+		return 0, response.WrapErrorWithData(response.ErrInvalidParams, fmt.Errorf("failed to compute tx size: %w", err))
+	}
+	size := len(hashablePart) + io.GetVarSize(len(tx.Signers))
+	var (
+		ef     int64
+		netFee int64
+	)
+	for i, signer := range tx.Signers {
+		var verificationScript []byte
+		for _, w := range tx.Scripts {
+			if w.VerificationScript != nil && hash.Hash160(w.VerificationScript).Equals(signer.Account) {
+				// then it's a standard sig/multisig witness
+				verificationScript = w.VerificationScript
+				break
+			}
+		}
+		if verificationScript == nil { // then it still might be a contract-based verification
+			res, respErr := s.runScriptInVM(trigger.Verification, []byte{}, signer.Account, tx)
+			if respErr != nil && errors.Is(respErr.Cause, core.ErrUnknownVerificationContract) {
+				// it's neither a contract-based verification script nor a standard witness attached to
+				// the tx, so the user did not provide enough data to calculate fee for that witness =>
+				// we don't take it into account.
+				continue // TODO: @roman-khimov, I'd better return an error in this case, because
+				// 1) the user might expect that RPC node wallet contains the proper account with verification script (we don't have the wallet here => error)
+				// 2) the user might just forgot to add the proper verification script to the transaction witnesses (it's a user error => we need to notify about it)
+				// but C# code continues here, see https://github.com/neo-project/neo/blob/master/src/neo/Wallets/Wallet.cs#L551
```

Comment:
I'd file it as a bug for C# code.

Reply:
https://github.com/neo-project/neo/pull/2413

Reply:
Also implemented as proposed to Neo devs.